pax/plugins/peak_processing/PeakAreaCorrections.py

```diff
@@ -64,3 +69,223 @@ def transform_event(self, event):
+                wf_sum_saturate=np.zeros_like(wf_sum_all) # Sum of non-saturated
+                for sample in range(start_sample,end_sample):
+                    for pmtid in range(0, nPmts):
+                        # first deal with non-saturated PMTS
+                        if(channel_sat_samples[pmtid]<1):
+                            wf_sum_nan_saturate[sample-start_sample]+=wf_raw[pmtid,sample-start_sample]
+                        else:
+                            wf_sum_saturate[sample-start_sample]+=wf_raw[pmtid,sample-start_sample]
+                area_nan_saturate=np.sum(wf_sum_nan_saturate) # Area of Total non-saturated WFs 
+                # The apply correction on saturated samples, need the shape of non-saturated WFs
+                # re-define pulse shape based on the derived model data
+                max_sample=np.argmax(wf_sum_nan_saturate) #sample sample at maximum
+                height=wf_sum_nan_saturate[max_sample]
+                peak_threshold_left=0.01*height
+                peak_threshold_right=0.02*height
+                
+                # search for more precise peak left/right edges
+                peak_left_edge=max_sample
+                for sample in reversed(range(start_sample,max_sample+start_sample)):
+                    if(wf_sum_nan_saturate[sample-start_sample]<=peak_threshold_left):
+                        peak_left_edge=sample
+                        break
+                if(peak_left_edge==max_sample): # use old edges if new edges now found
+                    peak_left_edge=start_sample 
+                    
+                peak_right_edge=max_sample
+                for sample in range(max_sample+start_sample,end_sample):
+                    if(wf_sum_nan_saturate[sample-start_sample]<=peak_threshold_right):
+                        peak_right_edge=sample
+                        break
+                if(peak_right_edge==max_sample): # use old edges if new edges now found
+                    peak_right_edge=end_sample
+
+                
+                # re-calculate area in the new window
+                area_nan_saturate=0.0
+                peak_left_edge=np.maximum(peak_left_edge,start_sample)
+                peak_right_edge=np.minimum(peak_right_edge,end_sample)
+                for sample in range(peak_left_edge,peak_right_edge):
+                    area_nan_saturate+=wf_sum_nan_saturate[sample-start_sample]
+                
+                # Area per pmt w/o saturation correction
+                area_pmt_before_correction=np.sum(wf_raw,axis=1)
+                area_pmt_after_correction=np.zeros(nPmts)
+                area_pmt_correction_factor=np.zeros(nPmts)
+                   
+                areatop=0.0
+                for pmtid in range(0,nPmts):
+                    area_model_left,area_model_right=0.0,0.0
+                    area_data_left,area_data_right=0.0,0.0
+                    #for sample in range(start_sample,end_sample):
```

Comment:
__[flake8]__
* [E265] block comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->